ios/RNMagicScript/components/UiNodes/Toggle/UiToggleGroupNode.swift

```diff
@@ -42,87 +43,66 @@ import SceneKit
-        addToggleNode(toggleNode)
-    }
-
-    fileprivate func registerToggleGroupHandler(_ node: UiToggleNode) {
-        node.onChangeGroup = { [weak self] node in
-            self?.toggleSelection(node)
-        }
-    }
-
-    func toggleSelection(_ node: UiToggleNode) {
-        if allowMultipleOn && allowAllOff {
-            node.on = !node.on
-        } else if allowMultipleOn && allowAllOff == false {
-            let filteredItems = itemsList.filter { $0.on }
-            if filteredItems.count == 1 && filteredItems.first == node { return }
-            node.on = !node.on
-        } else if allowMultipleOn == false && allowAllOff {
-            let filteredItems = itemsList.filter { $0.on }
-            if filteredItems.count >= 1 && filteredItems.contains { $0 != node }  { filteredItems.forEach { $0.on = false } }
-            node.on = !node.on
-        } else {
-            let filteredItems = itemsList.filter { $0 != node }
-            filteredItems.forEach { $0.on = false }
-            node.on = true
+        if innerGroup == nil && innerLayout == nil && child is TransformNodeContainer {
+            innerLayout = child
         }
-    }
 
-    fileprivate func addToggleNode(_ toggleNode: UiToggleNode) {
-        itemsList.append(toggleNode)
+        contentNode.addChildNode(child)
         setNeedsLayout()
+
+        return true
     }
 
     @objc override func removeChild(_ child: TransformNode) {
         if let toggleItem = child as? UiToggleNode {
-            itemsList.removeAll { node -> Bool in
-                return node == toggleItem
+            itemsList.removeAll { $0 == toggleItem }
+            if itemsList.isEmpty {
+                innerGroup = nil
             }
+            setNeedsLayout()
         }
 
         if let storedInnerLayout = self.innerLayout, storedInnerLayout == child {
             storedInnerLayout.removeFromParentNode()
-            self.innerLayout = nil
+            innerLayout = nil
+            setNeedsLayout()
         }
 
-        setNeedsLayout()
+        super.removeChild(child)
     }
 
     override func hitTest(ray: Ray) -> TransformNode? {
-         if innerLayout != nil {
-            return self.innerLayout?.hitTest(ray: ray)
-        }
-
-        for item in itemsList {
-            let hitNode = item.hitTest(ray: ray)
-            if hitNode != nil { return hitNode }
+        guard let _ = selfHitTest(ray: ray) else { return nil }
+        if let layout = innerLayout, let result = layout.hitTest(ray: ray) {
+            return result
+        } else if let group = innerGroup, let result = group.hitTest(ray: ray) {
+            return result
         }
 
-        return selfHitTest(ray: ray)
+        return self
     }
 
     @objc override func _calculateSize() -> CGSize {
-        return innerLayout?.getSize() ?? CGSize.zero
+        if innerLayout != nil {
+            return innerLayout!.getSize()
+        } else if innerGroup != nil {
```

Comment:
Note: I think we should revisit this part and try to discover more self-descriptive names (other than innerLayout & innerGroup)- at this moment it confuses me a little bit which is which.

Reply:
After analyzing the case, I suggest the following names:
innerLayout -> **customNodeContainer**
innerGroup -> **defaultNodeContainer**